File: src/main/kotlin/com/elissamc/core/UpdatedParts.java
package com.elissamc.core;

import jline.internal.Log;

import java.util.ArrayList;
import java.util.List;

/**
 * Class that turns off all parts when needed. Idk what is this...
 *
 * @author Mato Kormuth
 *
 */
public class UpdatedParts {
    private static List<Updatable> parts = new ArrayList<Updatable>();

    public static void shutdown() {
        for (Updatable part : UpdatedParts.parts)
            part.updateStop();
        UpdatedParts.parts.clear();
    }

    public static void registerPart(final Updatable part) {
        Log.info("[UpdatedParts] Registering: " + part.getClass().getSimpleName());
        UpdatedParts.parts.add(part);
    }
}

File: src/main/kotlin/com/elissamc/core/UpdatedParts.kt
package com.elissamc.core

import jline.internal.Log

import java.util.ArrayList

/** Class that turns off all parts when needed. */
object UpdatedParts {
    private val parts = ArrayList<Updatable>()

    fun shutdown() {
        for (part in UpdatedParts.parts)
            part.updateStop()
        UpdatedParts.parts.clear()
    }

    fun registerPart(part: Updatable) {
        Log.info("[UpdatedParts] Registering: " + part.javaClass.simpleName)
        UpdatedParts.parts.add(part)
    }
}